### PR TITLE
[VitisAI] fix graph save

### PR DESCRIPTION
**Pull request by BoarQing**
### Description

Revert the wrong change in https://github.com/microsoft/onnxruntime/pull/20920

### Motivation and Context

It would save the data at a wrong position

### CODE REVIEW

File: onnxruntime/core/providers/vitisai/imp/graph.cc [hunk 1]
```diff
@@ -107,12 +107,11 @@ void graph_save(const Graph& graph, const std::string& filename, const std::stri
   auto graph_proto_subgraph = graph.ToGraphProto();
   *model_proto->mutable_graph() = *graph_proto_subgraph;
   auto& logger = logging::LoggingManager::DefaultLogger();
-  auto filename_data_relative_path = std::filesystem::path();
   auto model = Model::Create(std::move(*model_proto), ToPathString(filename), nullptr, logger);
   if (initializer_size_threshold == std::numeric_limits<size_t>::max()) {
     model_proto = model->ToProto();
   } else {
-    model_proto = model->ToGraphProtoWithExternalInitializers(filename_dat, graph.ModelPath(), initializer_size_threshold);
+    model_proto = model->ToGraphProtoWithExternalInitializers(filename_dat, ToPathString(filename), initializer_size_threshold);
```
Comment (snnn) [inline onnxruntime/core/providers/vitisai/imp/graph.cc:114]: The first and the second parameters of the ToGraphProtoWithExternalInitializers function has the same type. 
So, here you also need to call ToPathString for filename_dat. This is based on an assumption that `filename_dat` and `filename` are encoded in UTF-8.

**Comment by snnn** [comment]
>   bool result = model_proto->SerializeToOstream(output);

Here you cannot pass filename to a std function, as we assume all multibyte strings are utf-8 strings.
A std function expect a string in native encoding, i.e. ISO8859-1, CP936 etc. 
So, here you also need to call ToPathString. Or, you may change this function's signature to use a different type for `filename`.

---
Refers to: onnxruntime/core/providers/vitisai/imp/graph.cc:127 in b1d0023. [](commit_id = b1d0023de1a8801b31e44857ba085ba3365b21e9, deletion_comment = False)

File: onnxruntime/core/providers/vitisai/imp/graph.cc (hunk 1 above)
Comment (BoarQing) [reply onnxruntime/core/providers/vitisai/imp/graph.cc]: Hi, I have changed everything you requested. Could you take a look. @snnn

**Comment by BoarQing** [comment]
Hi, I fixed the conflict. Could you start the pipeline @snnn

**Comment by BoarQing** [comment]
@HectorSVC Could you take a look. This is required for the release.

**Review by HectorSVC — APPROVED** [review]

**Pull request merged**